File: lista_de_exercicios/questao5/questao5.py
class Animal:
    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return "Latido"

class Gato(Animal):
    def som(self):
        return "Miau"

animais = [Cachorro(), Gato()]

for animal in animais:
    print(animal.som())

File: lista_de_exercicios/questao5/questao5.go
package main

import "fmt"

type Animal interface {
    Som() string
}

type Cachorro struct{}

func (C Cachorro) Som() string {
    return "Latido"
}

type Gato struct{}

func (G Gato) Som() string {
    return "Miau"
}

func main() {
    animais := []Animal{Cachorro{}, Gato{}}

    for _, animal := range animais {
        fmt.Println(animal.Som())
    }
}
